src/KeyStoreUtils.ts

```diff
@@ -57,8 +59,17 @@ export namespace KeyStoreUtils {
             if (!bip39.validateMnemonic(mnemonic)) { throw new Error('The given mnemonic could not be validated.'); }
         }
 
-        const seed = (await bip39.mnemonicToSeed(mnemonic, password)).slice(0, 32);
-        const keys = await generateKeys(seed);
+        let keys: { secretKey: Buffer, publicKey: Buffer };
+        const seed = await bip39.mnemonicToSeed(mnemonic, password);
+        if (derivationPath !== undefined) {
+            const sk = (Ed25519.derivePath(derivationPath, seed.toString("hex"))).key;
```

Comment:
why do you need both sk and p? seems sk isnt used

Reply:
Thank you